code/game/objects/machinery/vending/vending.dm

```diff
@@ -610,63 +629,148 @@
+						return FALSE
+
+					else if(machine_current_charge < recharge_amount) // Not enough but some charge remaining so partially recharge cell and move on
+						cell.give(machine_current_charge)
+						machine_current_charge = 0
+						cell.update_icon()
+						display_message_and_visuals(user, show_feedback, "Cell charged partially! [round(cell.percent())]%.", VENDING_RESTOCK_RECHARGE)
+						playsound(loc, 'sound/machines/hydraulics_1.ogg', 25, 0, 1)
+						return FALSE
+
+				else
+					display_message_and_visuals(user, show_feedback, "Cell isn't at full charge!", VENDING_RESTOCK_DENY)
+					return FALSE
+
+		if(isitemstack(item_to_stock))
 			var/obj/item/stack/stack = item_to_stock
 			if(stack.amount != initial(stack.amount))
-				to_chat(user, span_warning("[stack] has been partially used. You must replace the missing amount before you can restock it."))
-				return
-		else if(istype(item_to_stock, /obj/item/reagent_containers))
+				display_message_and_visuals(user, show_feedback, "[stack] has been partially used. Refill it!", VENDING_RESTOCK_DENY)
+				return FALSE
+
+		if(isreagentcontainer(item_to_stock))
 			var/obj/item/reagent_containers/reagent_container = item_to_stock
 			if(!reagent_container.free_refills && !reagent_container.has_initial_reagents())
-				to_chat(user, span_warning("\The [reagent_container] is missing some of its reagents. You must refill it before you can restock it."))
-				return
+				display_message_and_visuals(user, show_feedback, "\The [reagent_container] is missing some of its reagents!", VENDING_RESTOCK_DENY)
+				return FALSE
+
+	// At this point the item is guaranteed to be accepted into the vending machine
+
+	if(item_to_stock.loc == user) //Inside the mob's inventory
+		if(item_to_stock.flags_item & WIELDED)
+			item_to_stock.unwield(user)
+		user.transferItemToLoc(item_to_stock, src)
+
+	if(istype(item_to_stock.loc, /obj/item/storage)) //inside a storage item
+		var/obj/item/storage/S = item_to_stock.loc
+		S.remove_from_storage(item_to_stock, user.loc, user)
+
+	if(vending_flags & VENDING_RECHARGER && recharge_amount)
+		machine_current_charge -= recharge_amount
+		display_message_and_visuals(user, show_feedback, "Restocked and recharged", VENDING_RESTOCK_ACCEPT_RECHARGE)
+	else
+		display_message_and_visuals(user, show_feedback, "Restocked", VENDING_RESTOCK_ACCEPT)
+
+	qdel(item_to_stock)
+
+	if(record.amount >= 0) //R negative means infinite item, no need to restock
+		record.amount++
+
+	updateUsrDialog()
+	return TRUE //Item restocked, no reason to go on.
+
+/// Vending machine tries to restock all of the loose item on it's location onto itself.
+/obj/machinery/vending/proc/stock_vacuum(mob/user)
+	var/stocked = FALSE
+
+	for(var/obj/item/I in loc)
+		stocked = stock(I, null, FALSE) ? TRUE : stocked
+
+	stocked ? display_message_and_visuals(user, TRUE, "Automatically restocked all items from outlet.", VENDING_RESTOCK_ACCEPT) : null
+	restock_timer ? deltimer(restock_timer) : null
```

Comment:
is there some reason these aren't just ifs?

Reply:
Personal preferences if i just add a single call after a condition. If it's more i just do the usual if() indentation